DomainContent/Zaru/applause-app/applause-app.js

```diff
@@ -0,0 +1,420 @@
+            names.push(name);
+        }
+        return names;
+    }
+
+    function makeOpenPalm() {
+        ['right', 'left'].forEach(function(side) {
+
+            for (var i = 0; i < fingerKeys.length; i++) {
+                var finger = fingerKeys[i];
+                var jointSuffixes = 3; // We need to update rotation of the 0, 1 and 2 joints
+                var names = getJointNames(side, finger, jointSuffixes);
+
+                // update every finger joint
+
+                for (var j = 0; j < names.length; j++) {
+                    var index = MyAvatar.getJointIndex(names[j]);
+                    // if no finger is touching restate the default poses
+                    var quatRot = dataOpen[side][finger][j];
+                    MyAvatar.setJointRotation(index, quatRot);
+                }
+            }
+        });
+    }
+
+    function clearJoints() {
+        ['right', 'left'].forEach(function(side) {
+
+            for (var i = 0; i < fingerKeys.length; i++) {
+                var finger = fingerKeys[i];
+                var jointSuffixes = 3; // We need to update rotation of the 0, 1 and 2 joints
+                var names = getJointNames(side, finger, jointSuffixes);
+
+                // update every finger joint
+
+                for (var j = 0; j < names.length; j++) {
+                    var index = MyAvatar.getJointIndex(names[j]);
+                    // if no finger is touching restate the default poses
+                    MyAvatar.clearJointData(index);
+                }
+            }
+        });
+    }
+
+    function compareRotations(q1, q2) {
+        var threshold = 0.85;
+        return (Quat.dot(q1, q2) <= threshold);
+    }
+
+    function getRandomApplauseSound() {
+        return INDIVIDUAL_CLAP_URLS[Math.round(Math.random() * INDIVIDUAL_CLAP_URLS.length - 1)];
+    }
+
+    function checkHandsDistance() {
+
+        var handPositionR = MyAvatar.getJointPosition('RightHand');
+        var handPositionL = MyAvatar.getJointPosition('LeftHand');
+        var handRotationR = MyAvatar.getRightPalmRotation();
+        var handRotationL = MyAvatar.getLeftPalmRotation();
+        var oldRotationR = previousHandOrientations[0];
+        var oldRotationL = previousHandOrientations[1];
+
+        if (Vec3.distance(handPositionL, handPositionR) <= handProximityDistance) {
+            if (!handsStretched) {
+                makeOpenPalm();
+                handsStretched = true;
+            }
+        } else {
+            if (handsStretched) {
+                clearJoints();
+                handsStretched = false;
+            }
+        }
+        if ((Vec3.distance(handPositionL, handPositionR) <= handProximityDistance) 
+            && (compareRotations(oldRotationR, handRotationR) 
+            || compareRotations(oldRotationL, handRotationL)) && applauseEnabled) {
+            playSingleClap();
+            previousHandOrientations = [];
+            previousHandOrientations.push(handRotationR);
```

Comment:
I think I want to keep them in the array since they're always tied together and align with the other ways we can access controller and hand properties, and it's easier to clear them together instead of doubling the operations for separate variables.